flink-ml-lib/src/test/java/org/apache/flink/ml/feature/MinMaxScalerTest.java

```diff
@@ -0,0 +1,204 @@
+                output.getResolvedSchema().getColumnNames());
+    }
+
+    @Test
+    public void testFewerDistinctPointsThanCluster() {
+        MinMaxScaler minMaxScaler = new MinMaxScaler();
+        MinMaxScalerModel model = minMaxScaler.fit(predictData);
+        model.transform(predictData);
+    }
+
+    @Test
+    public void testFitAndPredict() throws Exception {
+        MinMaxScaler minMaxScaler = new MinMaxScaler();
+        MinMaxScalerModel minMaxScalerModel = minMaxScaler.fit(trainData);
+        Table output = minMaxScalerModel.transform(predictData)[0];
+        verifyPredictionResult(output, minMaxScaler.getPredictionCol());
+    }
+
+    @Test
+    public void testSaveLoadAndPredict() throws Exception {
+        MinMaxScaler minMaxScaler = new MinMaxScaler();
+        MinMaxScaler loadedMinMaxScaler =
+                StageTestUtils.saveAndReload(
+                        env, minMaxScaler, tempFolder.newFolder().getAbsolutePath());
+        MinMaxScalerModel minMaxScalerModel = loadedMinMaxScaler.fit(trainData);
+        minMaxScalerModel =
+                StageTestUtils.saveAndReload(
+                        env, minMaxScalerModel, tempFolder.newFolder().getAbsolutePath());
+        assertEquals(
+                Arrays.asList("minVector", "maxVector"),
+                minMaxScalerModel.getModelData()[0].getResolvedSchema().getColumnNames());
+        Table output = minMaxScalerModel.transform(predictData)[0];
+        verifyPredictionResult(output, minMaxScaler.getPredictionCol());
+    }
+
+    @Test
+    public void testModelSaveLoadAndPredict() throws Exception {
+        MinMaxScaler minMaxScaler = new MinMaxScaler();
+        MinMaxScalerModel minMaxScalerModel = minMaxScaler.fit(trainData);
+        MinMaxScalerModel newModel =
+                StageTestUtils.saveAndReload(
+                        env, minMaxScalerModel, tempFolder.newFolder().getAbsolutePath());
+        Table output = newModel.transform(predictData)[0];
+        verifyPredictionResult(output, minMaxScaler.getPredictionCol());
+    }
+
+    @Test
+    public void testGetModelData() throws Exception {
+        MinMaxScaler minMaxScaler = new MinMaxScaler();
+        MinMaxScalerModel minMaxScalerModel = minMaxScaler.fit(trainData);
+        Table modelData = minMaxScalerModel.getModelData()[0];
+        DataStream<Row> output = tEnv.toDataStream(modelData);
+        assertEquals("minVector", modelData.getResolvedSchema().getColumnNames().get(0));
+        assertEquals("maxVector", modelData.getResolvedSchema().getColumnNames().get(1));
+        List<Row> modelRows = IteratorUtils.toList(output.executeAndCollect());
+        MinMaxScalerModelData data =
+                new MinMaxScalerModelData(
+                        (DenseVector) modelRows.get(0).getField(0),
+                        (DenseVector) modelRows.get(0).getField(1));
+        Assert.assertNotNull(data);
+        assertEquals(data.maxVector.size(), data.maxVector.size());
```

Comment:
Could you please also check the value of model data?

Reply:
OK